Assets/MixedRealityToolkit-Preview/LightEstimation/Scripts/Utility/CubeMapper.cs

```diff
@@ -0,0 +1,367 @@
+		
+			// Figure out which faces of the Cubemap we need to render with this stamp, we don't generally need to 
+			// render all of them, (usually 3 at most) so it's a no-brainer optimization, even if it's not 100% accurate
+			int     render = 0;
+			Vector3 fwd    = cam.transform.forward;
+			if (needsFullStamp)
+			{
+				render = 63;
+			}
+			else
+			{
+				if (AngleVisible(fwd,  Vector3.right  )) render |= 1 << (int)CubemapFace.PositiveX;
+				if (AngleVisible(fwd, -Vector3.right  )) render |= 1 << (int)CubemapFace.NegativeX;
+				if (AngleVisible(fwd,  Vector3.up     )) render |= 1 << (int)CubemapFace.PositiveY;
+				if (AngleVisible(fwd, -Vector3.up     )) render |= 1 << (int)CubemapFace.NegativeY;
+				if (AngleVisible(fwd,  Vector3.forward)) render |= 1 << (int)CubemapFace.PositiveZ;
+				if (AngleVisible(fwd, -Vector3.forward)) render |= 1 << (int)CubemapFace.NegativeZ;
+			}
+
+			// render the objects
+			root.SetActive(true);
+			Material old = RenderSettings.skybox;
+			RenderSettings.skybox = skybox;
+			cam.RenderToCubemap(map, render);
+			RenderSettings.skybox = old;
+			root.SetActive(false);
+		
+			CacheStamp(aPosition, aOrientation);
+			needsFullStamp = false;
+		}
+		public void Clear()
+		{
+			needsFullStamp = true;
+			ClearCache();
+		}
+		public void ClearCache()
+		{
+			stampCache.Clear();
+		}
+		public bool IsCached(Vector3 position, Quaternion orientation)
+		{
+			float min = float.MaxValue;
+
+			for (int i = 0; i < stampCache.Count; i++)
+			{
+				StampDir s = stampCache[i];
+				if  (IsExpired(i, position))
+				{
+					stampCache.RemoveAt(i);
+					continue;
+				}
+				float ang = Quaternion.Angle(orientation, s.direction);
+				if (min > ang)
+					min = ang;
+			}
+
+			// allow for a little overlap (using .4 fov instead of .5), especially since fov is horiontal, not vertical, and vertical is 
+			// shorter due to aspect ratio
+			return min < fov*(9f/16f);
+		}
+		#endregion
+
+		#region Private Methods
+		private Mesh CreatePlane(float aZ, float aBlend = 0.2f)
+		{
+			Mesh m = new Mesh();
+			m.vertices = new Vector3[] { 
+				new Vector3(-1,1,aZ), new Vector3(1,1,aZ), new Vector3(1,-1,aZ), new Vector3(-1,-1,aZ),
+				new Vector3(-1 + aBlend, 1-aBlend*2, aZ), new Vector3(1-aBlend, 1-aBlend*2, aZ), new Vector3(1-aBlend, -1+aBlend*2, aZ), new Vector3(-1+aBlend, -1+aBlend*2, aZ) };
+			m.colors = new Color[] { new Color(1,1,1,0), new Color(1,1,1,0), new Color(1,1,1,0), new Color(1,1,1,0),
+				Color.white, Color.white, Color.white, Color.white };
+			m.uv = new Vector2[] { new Vector2(0, 1), new Vector2(1,1), new Vector2(1,0), new Vector2(0,0),
+				new Vector2(aBlend/2, 1-aBlend), new Vector2(1-aBlend/2, 1-aBlend), new Vector2(1-aBlend/2, aBlend), new Vector2(aBlend/2, aBlend)};
+			m.triangles = new int[] {
+				0,1,5,  0,5,4,
+				5,1,2,  5,2,6,
+				6,2,7,  7,2,3,
+				7,3,0,  7,0,4,
+				4,5,6,  4,6,7};
+			m.RecalculateBounds();
+
+			return m;
+		}
+		private Mesh CreateNautilusSphere()
```

Comment:
I've never heard of a Nautilus Sphere? Could you comment more about what it is/used for?

Reply:
I'll have a comment in the function to elaborate! Here's the details:

This function creates a sphere mesh with texture overlap at the back. It contracts as its vertices are generated, making it look kinda like the shell of a nautilus! The first vertical slice will be scaled by 1, and the last by 0.9 in this case. Since there's overlap on the last few slices, this ensures none of the geometry will zfight.
More careful geometry and shaders might not need this scaling, but it doesn't hurt even then, since the view from the center is the same regardless!